dts/arm/st/u0/stm32u0.dtsi

```diff
@@ -273,6 +307,115 @@
 			status = "disabled";
 		};
 
+		dma1: dma@40020000 {
+			compatible = "st,stm32-dma-v2";
+			#dma-cells = <3>;
+			reg = <0x40020000 0x400>;
+			interrupts = <9 0 10 0 11 0>;
+			clocks = <&rcc STM32_CLOCK(AHB1, 0U)>;
+			dma-channels = <7>;
+			dma-offset = <0>;
+			dma-requests= <12>;
```

Comment:
When using stm32-dma-v2, I think it is not important to specify `dma-requests` since requests are handled in DMAMUX.

Reply:
I took a look at your remark, and for me, there will be no issues because if we don't specify, the value 0 is given, which is not bad for this version of dma i think.

Besides, there is a confusion I found only for the `stm32-dma-v2` between `dma-requests` and `dma-channels`. It would have made more sense to me if it was only `dma-channels`. I think that work will be done later to avoid the confusion.

Reply:
I see, but I remember that I had issues when I removed it through testing in past.
will try again to remove it and check what happens

Reply:
When I removed dma-requests, I encountered issues.
I believe you intended me to remove dma-channels instead of dma-requests, similar to how it's handled on other boards.

I noticed that boards like stm32wb, stm32g0, stm32c0, and stm32wl do not mention dma-channels under the DMA nodes in their DTS configurations. 
To maintain consistency with these boards, I removed dma-channels and added dma-requests instead."